manager/apis/app/v1alpha1/m4dapplication_types.go

```diff
@@ -8,18 +8,24 @@ import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
-// DataContext indicates data set chosen by the Data Scientist to be used by his application,
+// DataContext indicates data set chosen by the Data Scientist to be read from or written to by his application,
 // and includes information about the data format and technologies used by the application
 // to access the data.
 type DataContext struct {
-	// DataSetID is a unique identifier of the dataset chosen from the data catalog for processing by the data user application.
+	// DataSetID is a unique identifier of the dataset either chosen from the enterprise data catalog for processing by the data user application,
+	// or the source location of a new data set being imported into the environment.  This parameter should contain datacatalogservice, catalog, datasetID.
 	// +required
 	// +kubebuilder:validation:MinLength=1
 	DataSetID string `json:"dataSetID"`
 
 	// IFdetails indicates the protocol and format expected by the data by the Data Scientist's application
 	// +required
 	IFdetails InterfaceDetails `json:"ifDetails"`
+
+	// DestinationCatalog indicates the catalog service and catalog in which a data set being imported should be registered.
+	// This parameter should contain datacatalogservice, and catalog.  This should be populated only when Cataloged is false.
+	// +optional
+	DestinationCatalog string `json:"destinationCatalog,omitempty"`
```

Comment:
I don't follow the structure. In the ingest case, why the asset will be under read?

Reply:
Removed to avoid confusion with flow

```yaml
selector: ...
appInfo: ...
data:
  - asset: ...
    preferences:
      protocol: ...
      format: ...
      copy:
        catalog: ... # effectively an explicit copy
        ...
```